app/infrastructure/cache/redis_cache.py

```
@@ -117,7 +143,16 @@ async def set(self, key: str, value: Any, ttl: int = 300) -> bool:
             await self.redis.setex(key, ttl, serialized)
             logger.debug("Cache SET: %s (TTL: %ss)", key, ttl)
             return True
-        except Exception:
+        except (redis.ConnectionError, redis.TimeoutError):
+            if await self._reconnect():
+                try:
+                    await self.redis.setex(key, ttl, serialized)
+                    return True
```

Ruff (0.15.0)

[warning] Consider moving this statement to an `else` block

(TRY300)